lib/HTTP/Response.pm

```diff
@@ -116,75 +116,102 @@ sub redirects {
     return reverse @r;
 }
 
-
-sub filename
-{
+sub filename {
     my $self = shift;
     my $file;
 
     my $cd = $self->header('Content-Disposition');
     if ($cd) {
-	require HTTP::Headers::Util;
-	if (my @cd = HTTP::Headers::Util::split_header_words($cd)) {
-	    my ($disposition, undef, %cd_param) = @{$cd[-1]};
-	    $file = $cd_param{filename};
-
-	    # RFC 2047 encoded?
-	    if ($file && $file =~ /^=\?(.+?)\?(.+?)\?(.+)\?=$/) {
-		my $charset = $1;
-		my $encoding = uc($2);
-		my $encfile = $3;
-
-		if ($encoding eq 'Q' || $encoding eq 'B') {
-		    local($SIG{__DIE__});
-		    eval {
-			if ($encoding eq 'Q') {
-			    $encfile =~ s/_/ /g;
-			    require MIME::QuotedPrint;
-			    $encfile = MIME::QuotedPrint::decode($encfile);
-			}
-			else { # $encoding eq 'B'
-			    require MIME::Base64;
-			    $encfile = MIME::Base64::decode($encfile);
-			}
-
-			require Encode;
-			require Encode::Locale;
-			Encode::from_to($encfile, $charset, "locale_fs");
-		    };
-
-		    $file = $encfile unless $@;
-		}
-	    }
-	}
+        require HTTP::Headers::Util;
+        if ( my @cd = HTTP::Headers::Util::split_header_words($cd) ) {
+            my ( $disposition, undef, %cd_param ) = @{ $cd[-1] };
+
+            $file = $cd_param{filename};
+
+            my ( $charset, $encoding, $lang, $encfile );
```

Comment:
there is no need to predeclare these variables - you can declare them right in the conditional.

Reply:
I had it that way at first, but I also wanted to do the same for the other branch and doing that gave a warning that the variables were being masked. I'll rename them instead.